Tests/ReactiveSwiftTests/SignalProducerSpec.swift

```diff
@@ -1962,6 +1962,59 @@ class SignalProducerSpec: QuickSpec {
 				let result = producer.single()
 				expect(result?.value) == 1
 			}
+			
+			context("with interval") {
+				
+				it("should send values at the given interval until hitting the limitation") {
+					
+					let scheduler = TestScheduler()
+					var count = 0
+
+					let original = SignalProducer<Int, TestError> { observer, _ in
+						
+						if count < 2 {
+							scheduler.schedule { observer.send(value: count) }
+							scheduler.schedule { observer.send(error: .default) }
+						} else {
+							scheduler.schedule { observer.sendCompleted() }
+						}
+						count += 1
+					}
+
+					var values: [Int] = []
+					var completed = false
+					
+					original.retry(upTo: Int.max, interval: 1, on: scheduler)
```

Comment:
I added a test that tests hitting the ``upto`` limit.
please see 
https://github.com/ReactiveCocoa/ReactiveSwift/pull/496/files#diff-9cd786f9e52e786771d96cf8c42d3befR2018